src/data/finsupp.lean

```diff
@@ -1446,6 +1446,10 @@ ext $ λ a', by by_cases a = a';
   [{ subst h, simp only [smul_apply', single_eq_same] },
    simp only [h, smul_apply', ne.def, not_false_iff, single_eq_of_ne, smul_zero]]
 
+@[simp] lemma smul_single_self {R : semiring γ}
```

Comment:
I'm confused by this name. Does the `self` refer to the fact that `c` and `b` have the same type?

Reply:
Yes, but I agree it's not good naming. I think I'll just call this `smul_single'`.